chore: update css-declaration-sorter to v6.0.3

Siilwyn opened the pull request:
Update so Node.js 10 is supported in combination with PostCSS 8.
https://github.com/cssnano/cssnano/pull/1065#issuecomment-835400711

Siilwyn commented:
Seems like some prefixed properties like `-webkit-text-size-adjust` and other properties like `text-rendering` shifted around. :thinking:

ludofischer commented:
> Seems like some prefixed properties like `-webkit-text-size-adjust` and other properties like `text-rendering` shifted around. thinking

I think it would be a good idea to add some tests to https://github.com/cssnano/cssnano/blob/master/packages/cssnano-preset-default/src/__tests__/css-declaration-sorter.js, for example to check that integrating the new `css-declaration-sorter` release fixes https://github.com/cssnano/cssnano/issues/1054

alexander-akait commented:
Agree :+1:

alexander-akait reviewed (commented):
Can we add test for `all`?

Siilwyn commented:
Rebased and added four tests, one test has weird output not sure what to do with it, maybe something to put into `knownIssues.js`?
```
    Expected: "a{border-top:1px solid;border-color:purple}"
    Received: "a{border-color:currentcolor purple purple;border-top:1px solid purple}"
```

alexander-akait commented:
Yep, weird output, maybe problem due order of plugin?

Siilwyn commented:
@alexander-akait I did try removing css-declaration-sorter but the test fails with the same output, maybe it's a bug in the merge-longhand logic?

alexander-akait commented:
> @alexander-akait I did try removing css-declaration-sorter but the test fails with the same output, maybe it's a bug in the merge-longhand logic?

let's skip it and open an issue for this :+1:

Siilwyn commented:
@alexander-akait alright, it's odd though if I run the same input in the cssnano playground it doesn't have this behaviour.

ludofischer commented on packages/cssnano-preset-default/src/__tests__/css-declaration-sorter.js:
```diff
@@ -34,9 +34,23 @@ test(
-  passthroughCSS(
-    'a{animation-name:a;animation:b;-webkit-animation:c;-webkit-animation-name:d}'
+  'keep order of border properties, short- and longhand, left to right',
+  passthroughCSS('a{border-color:purple;border-top:1px solid}')
```
I also had trouble with this test example, if you reduce it to the essential it fails because of some non-optimal `merge-lonhand` behaviour. I suggest you replace it with the exact example in the [original bug report](https://github.com/cssnano/cssnano/issues/1054): 
```suggestion
  processCSS(
    'a{border-color:purple;border-top-width:1px;border-top-style:solid}',
    'a{border-top:1px solid;border-color:purple}'
  )
```

alexander-akait commented:
Need fix CI

ludofischer commented:
> Need fix CI

Probably need to regenerate the integration tests

```
yarn build:integration
```

Siilwyn commented:
> > Need fix CI
> 
> Probably need to regenerate the integration tests
> 
> ```
> yarn build:integration
> ```

Yes as I said before:
> Also the framework integration tests keep failing now, even though I did run yarn build:integration.

So even after running this command there are no changes, any help appreciated.

alexander-akait commented:
/cc @ludofischer Can you look? Maybe we have problems with tests...

Siilwyn commented:
I rebased, and ran `yarn build:integration` just to be sure.

ludofischer commented:
> I rebased, and ran `yarn build:integration` just to be sure.

But did you commit the modified CSS fixtures? They're in `packages/cssnano-preset-default/src/__tests__/integrations/` and `packages/cssnano-preset-advanced/src/__tests__/integrations/`. I still see the integration tests failing on CI.

Siilwyn commented:
> > I rebased, and ran `yarn build:integration` just to be sure.
> 
> But did you commit the modified CSS fixtures? They're in `packages/cssnano-preset-default/src/__tests__/integrations/` and `packages/cssnano-preset-advanced/src/__tests__/integrations/`. I still see the integration tests failing on CI.

The CSS fixtures stay the same, so nothing to commit. Just to be 200% sure I removed all files in the directories you mention manually before running the build:integration script.

ludofischer commented:
> The CSS fixtures stay the same, so nothing to commit. Just to be 200% sure I removed all files in the directories you mention manually before running the build:integration script.

Do the tests pass locally? Because in the test logs the result does not match the fixture.

ludofischer commented:
I've opened https://github.com/cssnano/cssnano/pull/1071 to add a commit with the fixtures fixed. What I think happened is that the tests testthe code in `package/src` but the fixtures are generated with the code in `dist` so  you need to run `yarn` to rebuild all packages before running `build:integration`.

Siilwyn commented:
Alright let me know where to 'continue' this PR, I've cherry picked 4ec0f00. Can also close this one. Thanks for the help!
